src/components/MovieListPage/MovieListPage.tsx

```diff
@@ -0,0 +1,192 @@
+  const [total, setTotal] = useState<number>();
+
+  const fetchData = async (params?: MovieQueryParams) => {
+    try {
+      if (cancelSource) {
+        cancelSource.cancel();
+        setCancelSource(null);
+      }
+      const source = axios.CancelToken.source();
+      setCancelSource(source);
+      const result = await new MoviesService().get(params, source);
+      setCancelSource(null);
+      setMovieList(result.data);
+      setTotal(result.totalAmount);
+    } catch (error) {
+      if (axios.isCancel(error)) return;
+    }
+  };
+
+  const [dialogContent, setDialogContent] = useState<DialogProps>({
+    children: null,
+    title: '',
+  });
+
+  useEffect(() => {
+    fetchData({
+      search: searchTerm || genre,
+      searchBy: searchTerm ? 'title' : 'genres',
+      sortBy: sort,
+      sortOrder: 'asc',
+    });
+  }, [searchTerm, genre, sort]);
+
+  function handleMovieEdit(movie: Movie) {
+    setDialogContent({
+      title: 'EDIT MOVIE',
+      children: (
+        <MovieForm
+          movie={movie}
+          onSubmit={(mov) => {
+            setShowDialog(false);
+          }}
+        ></MovieForm>
+      ),
+    });
+    setShowDialog(true);
+  }
+
+  function handleMovieDeleteClick() {
+    setDialogContent({
+      title: 'DELETE MOVIE',
+      children: (
+        <DialogContent>
+          <span>Are you sure you want to delete this movie?</span>
+          <ConfirmButton onClick={() => setShowDialog(false)}>
+            CONFIRM
+          </ConfirmButton>
+        </DialogContent>
+      ),
+    });
+    setShowDialog(true);
+  }
+
+  function handleAddMovie() {
+    setDialogContent({
+      title: 'EDIT MOVIE',
+      children: (
+        <MovieForm
+          onSubmit={(mov) => {
+            setShowDialog(false);
+          }}
+        ></MovieForm>
+      ),
+    });
+    setShowDialog(true);
+  }
+
+  return (
+    <MovieListPageContainer>
+      <TopContainer>
+        {selectedMovie ? (
+          <DetailsContainer>
+            <TopContainerHeader>
+              <AppLogo />
+              <SearchSwitcherButton onClick={() => setSelectedMovie(undefined)}>
+                <Icon src={SearchLogoIcon} />
+              </SearchSwitcherButton>
+            </TopContainerHeader>
+            <MovieDetails movie={selectedMovie}></MovieDetails>
+          </DetailsContainer>
+        ) : (
+          <SearchMovieContainer>
+            <TopContainerHeader>
+              <AppLogo />
+              <AddMovieButton onClick={handleAddMovie}>
+                + ADD MOVIE
+              </AddMovieButton>
+            </TopContainerHeader>
+            <SearchMovieContent>
+              <SearchForm initialValue={searchTerm} onSearch={setSearchTerm} />
+            </SearchMovieContent>
+          </SearchMovieContainer>
+        )}
+      </TopContainer>
+
+      <MoviesContainer>
+        <DetailsHeader>
+          <GenreSelect genreList={GENRE_LIST} onSelect={setGenre} />
+          <SortControl
```

Comment:
1) Please fix tests for `SortControl.tests.tsx`.
To avoid such issues it is better to create a mock for sort options and then import and set interface as for SORT_OPTIONS - or simply use the same constant - `SORT_OPTIONS` - instead of creating extra constants
`const mockSortList = ['name', 'rating', 'year']; ` 
 just for tests

2) And stories for `SortControl`